File: generate_icon.py
#!/usr/bin/env python3
"""
FocusFlow App Icon Generator
生成与 UI 风格一致的现代化图标
"""

from PIL import Image, ImageDraw, ImageFilter
import math
import os

def create_gradient_orb(size, center_x, center_y, radius):
    """创建渐变球体"""
    orb = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(orb)
    
    # 创建径向渐变
    for r in range(radius, 0, -1):
        # 从橙色到蓝色的渐变
        ratio = r / radius
        if ratio > 0.5:
            # 外圈：橙色
            r_color = int(255 * (ratio - 0.5) * 2)
            g_color = int(107 * (ratio - 0.5) * 2)
            b_color = int(53 * (ratio - 0.5) * 2)
        else:
            # 内圈：蓝色
            r_color = int(79 + (255 - 79) * ratio * 2)
            g_color = int(172 + (107 - 172) * ratio * 2)
            b_color = int(254 + (53 - 254) * ratio * 2)
        
        alpha = int(150 * (r / radius))
        color = (r_color, g_color, b_color, alpha)
        
        draw.ellipse([
            center_x - r, center_y - r,
            center_x + r, center_y + r
        ], fill=color)
    
    return orb

def create_decorative_lines(size):
    """创建装饰细线"""
    lines = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(lines)
    
    # 曲线1
    points1 = []
    for t in range(0, 100):
        x = int(size * 0.1 + (size * 0.8) * t / 100)
        y = int(size * 0.3 + size * 0.2 * math.sin(t * 0.1))
        points1.append((x, y))
    
    for i in range(len(points1) - 1):
        draw.line([points1[i], points1[i+1]], fill=(255, 255, 255, 25), width=2)
    
    # 曲线2
    points2 = []
    for t in range(0, 100):
        x = int(size * 0.1 + (size * 0.8) * t / 100)
        y = int(size * 0.6 + size * 0.15 * math.sin(t * 0.08 + 1))
        points2.append((x, y))
    
    for i in range(len(points2) - 1):
        draw.line([points2[i], points2[i+1]], fill=(255, 255, 255, 15), width=1)
    
    return lines

def draw_timer_icon(draw, size, center_x, center_y, icon_size):
    """绘制计时器图标"""
    # 外圈
    outer_radius = icon_size // 2
    draw.ellipse([
        center_x - outer_radius, center_y - outer_radius,
        center_x + outer_radius, center_y + outer_radius
    ], outline=(255, 255, 255, 200), width=8)
    
    # 顶部按钮
    button_width = 16
    button_height = 12
    draw.rectangle([
        center_x - button_width // 2, center_y - outer_radius - button_height + 4,
        center_x + button_width // 2, center_y - outer_radius + 4
    ], fill=(255, 255, 255, 200))
    
    # 指针
    hand_length = int(outer_radius * 0.6)
    angle = -45  # 45度角
    rad = math.radians(angle)
    end_x = int(center_x + hand_length * math.cos(rad))
    end_y = int(center_y + hand_length * math.sin(rad))
    
    draw.line([
        (center_x, center_y),
        (end_x, end_y)
    ], fill=(255, 255, 255, 255), width=6)
    
    # 中心点
    draw.ellipse([
        center_x - 6, center_y - 6,
        center_x + 6, center_y + 6
    ], fill=(255, 255, 255, 255))

def generate_icon(size):
    """生成指定尺寸的图标"""
    # 创建深色背景
    icon = Image.new('RGBA', (size, size), (10, 10, 10, 255))
    
    # 添加渐变球体
    orb_size = int(size * 0.7)
    orb_x = size // 2
    orb_y = int(size * 0.45)
    orb = create_gradient_orb(size, orb_x, orb_y, orb_size // 2)
    icon = Image.alpha_composite(icon, orb)
    
    # 添加装饰线
    lines = create_decorative_lines(size)
    icon = Image.alpha_composite(icon, lines)
    
    # 添加计时器图标
    icon_size = int(size * 0.35)
    icon_center_x = size // 2
    icon_center_y = int(size * 0.48)
    
    # 创建图标层
    icon_layer = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    icon_draw = ImageDraw.Draw(icon_layer)
    draw_timer_icon(icon_draw, size, icon_center_x, icon_center_y, icon_size)
    
    icon = Image.alpha_composite(icon, icon_layer)
    
    # 转换为 RGB（移除 alpha）
    icon_rgb = Image.new('RGB', (size, size), (10, 10, 10))
    icon_rgb.paste(icon, mask=icon.split()[3])
    
    return icon_rgb

def main():
    """主函数"""
    # 需要生成的尺寸
    sizes = {
        '1024.png': 1024,
        '180.png': 180,
        '167 1.png': 167,
        '152.png': 152,
        '120.png': 120,
        '114.png': 114,
        '87.png': 87,
        '80 1.png': 80,
        '80.png': 80,
        '60.png': 60,
        '58 1.png': 58,
        '58.png': 58,
        '57.png': 57,
        '40 1.png': 40,
        '40.png': 40,
        '29.png': 29,
    }
    
    output_dir = '/Users/fireny/Desktop/Qoder/FocusFlow/Assets.xcassets/AppIcon.appiconset'
    
    print("开始生成 FocusFlow 图标...")
    print("=" * 50)
    
    for filename, size in sizes.items():
        print(f"生成 {size}x{size} -> {filename}")
        icon = generate_icon(size)
        icon.save(os.path.join(output_dir, filename), 'PNG')
    
    print("=" * 50)
    print("✅ 所有图标生成完成！")
    print("\n下一步：")
    print("1. 在 Xcode 中清理构建缓存: Product → Clean Build Folder")
    print("2. 重新构建项目")
    print("3. 在模拟器或真机上查看新图标")

if __name__ == '__main__':
    main()

File: generate_icon.swift
#!/usr/bin/env xcrun swift
import Foundation
import AppKit

// MARK: - 图标生成器
struct IconGenerator {
    
    static func generate(size: CGSize) -> NSImage {
        let image = NSImage(size: size)
        image.lockFocus()
        
        // 深色背景
        NSColor(calibratedRed: 0.039, green: 0.039, blue: 0.039, alpha: 1.0).set()
        NSRect(origin: .zero, size: size).fill()
        
        // 渐变球体
        drawGradientOrb(in: size)
        
        // 装饰线
        drawDecorativeLines(in: size)
        
        // 计时器图标
        drawTimerIcon(in: size)
        
        image.unlockFocus()
        return image
    }
    
    static func drawGradientOrb(in size: CGSize) {
        let orbSize = size.width * 0.85
        let centerX = size.width / 2
        let centerY = size.height / 2  // 完全居中
        
        let orbRect = NSRect(
            x: centerX - orbSize / 2,
            y: centerY - orbSize / 2,
            width: orbSize,
            height: orbSize
        )
        
        // 创建径向渐变
        let colors = [
            NSColor(calibratedRed: 1.0, green: 0.42, blue: 0.21, alpha: 0.6).cgColor,  // 橙色
            NSColor(calibratedRed: 0.31, green: 0.67, blue: 0.99, alpha: 0.6).cgColor,  // 蓝色
            NSColor.clear.cgColor
        ]
        
        let gradient = CGGradient(
            colorsSpace: CGColorSpaceCreateDeviceRGB(),
            colors: colors as CFArray,
            locations: [0.0, 0.5, 1.0]
        )!
        
        let context = NSGraphicsContext.current!.cgContext
        context.drawRadialGradient(
            gradient,
            startCenter: CGPoint(x: orbRect.midX, y: orbRect.midY),
            startRadius: 0,
            endCenter: CGPoint(x: orbRect.midX, y: orbRect.midY),
            endRadius: orbSize / 2,
            options: .drawsAfterEndLocation
        )
    }
    
    static func drawDecorativeLines(in size: CGSize) {
        let path = NSBezierPath()
        
        // 曲线1
        path.move(to: CGPoint(x: size.width * 0.1, y: size.height * 0.3))
        path.curve(
            to: CGPoint(x: size.width * 0.9, y: size.height * 0.5),
            controlPoint1: CGPoint(x: size.width * 0.3, y: size.height * 0.2),
            controlPoint2: CGPoint(x: size.width * 0.7, y: size.height * 0.4)
        )
        
        NSColor(white: 1.0, alpha: 0.1).setStroke()
        path.lineWidth = size.width * 0.002  // 动态线条宽度
        path.stroke()
        
        // 曲线2
        let path2 = NSBezierPath()
        path2.move(to: CGPoint(x: size.width * 0.1, y: size.height * 0.6))
        path2.curve(
            to: CGPoint(x: size.width * 0.9, y: size.height * 0.7),
            controlPoint1: CGPoint(x: size.width * 0.3, y: size.height * 0.8),
            controlPoint2: CGPoint(x: size.width * 0.7, y: size.height * 0.6)
        )
        
        NSColor(white: 1.0, alpha: 0.05).setStroke()
        path2.lineWidth = size.width * 0.001  // 更细的线条
        path2.stroke()
    }
    
    static func drawTimerIcon(in size: CGSize) {
        let iconSize = size.width * 0.55
        let center = CGPoint(x: size.width / 2, y: size.height / 2)  // 完全居中
        let radius = iconSize / 2
        
        // 根据图标尺寸动态计算线条粗细
        let circleLineWidth = size.width * 0.012  // 加粗外圈线条（原 0.008）
        let handLineWidth = size.width * 0.010     // 加粗指针线条（原 0.006）
        let dotSize = size.width * 0.018           // 增大中心点（原 0.012）
        let buttonWidth = size.width * 0.024       // 增大顶部按钮（原 0.016）
        let buttonHeight = size.width * 0.018      // 增大顶部按钮（原 0.012）
        
        // 外圈
        let circlePath = NSBezierPath(ovalIn: NSRect(
            x: center.x - radius,
            y: center.y - radius,
            width: iconSize,
            height: iconSize
        ))
        
        NSColor(white: 1.0, alpha: 0.8).setStroke()
        circlePath.lineWidth = circleLineWidth
        circlePath.stroke()
        
        // 顶部按钮（macOS 坐标系 Y 轴向上）
        let buttonOffset = size.width * 0.004  // 按钮与圆的间距
        let buttonRect = NSRect(
            x: center.x - buttonWidth / 2,
            y: center.y + radius + buttonOffset,  // 在圆的上方
            width: buttonWidth,
            height: buttonHeight
        )
        NSColor(white: 1.0, alpha: 0.8).setFill()
        buttonRect.fill()
        
        // 指针（增长长度）
        let handLength = radius * 0.75  // 增长指针（原 0.6）
        let angle = -45.0 * .pi / 180.0
        let endX = center.x + handLength * cos(angle)
        let endY = center.y + handLength * sin(angle)
        
        let handPath = NSBezierPath()
        handPath.move(to: center)
        handPath.line(to: CGPoint(x: endX, y: endY))
        
        NSColor.white.setStroke()
        handPath.lineWidth = handLineWidth
        handPath.lineCapStyle = .round
        handPath.stroke()
        
        // 中心点
        let dotPath = NSBezierPath(ovalIn: NSRect(
            x: center.x - dotSize / 2,
            y: center.y - dotSize / 2,
            width: dotSize,
            height: dotSize
        ))
        NSColor.white.setFill()
        dotPath.fill()
    }
}

// MARK: - 主函数
func main() {
    let outputDir = "/Users/fireny/Desktop/Qoder/FocusFlow/Assets.xcassets/AppIcon.appiconset"
    
    let sizes: [(name: String, size: CGFloat)] = [
        ("1024.png", 1024),
        ("180.png", 180),
        ("167 1.png", 167),
        ("152.png", 152),
        ("120.png", 120),
        ("114.png", 114),
        ("87.png", 87),
        ("80 1.png", 80),
        ("80.png", 80),
        ("60.png", 60),
        ("58 1.png", 58),
        ("58.png", 58),
        ("57.png", 57),
        ("40 1.png", 40),
        ("40.png", 40),
        ("29.png", 29)
    ]
    
    print("开始生成 FocusFlow 图标...")
    print("=" .padding(toLength: 50, withPad: "=", startingAt: 0))
    
    for (name, size) in sizes {
        print("生成 \(Int(size))x\(Int(size)) -> \(name)")
        
        let icon = IconGenerator.generate(size: CGSize(width: size, height: size))
        
        guard let tiffData = icon.tiffRepresentation,
              let bitmapImage = NSBitmapImageRep(data: tiffData),
              let pngData = bitmapImage.representation(using: .png, properties: [:]) else {
            print("  ❌ 失败")
            continue
        }
        
        let url = URL(fileURLWithPath: outputDir).appendingPathComponent(name)
        try? pngData.write(to: url)
    }
    
    print("=" .padding(toLength: 50, withPad: "=", startingAt: 0))
    print("✅ 所有图标生成完成！")
    print("\n下一步：")
    print("1. 在 Xcode 中: Product → Clean Build Folder (⇧⌘K)")
    print("2. 重新构建项目")
    print("3. 在模拟器或真机上查看新图标")
}

main()
